### ISSUE TITLE
systemtap package: incompatible with linux >= 4.6, needs some patches backported from upstream, cf. Debian bug report

### ISSUE THREAD

**Issue by ackalker** [voidlinux/void-packages#4905]
Related [Debian bug report](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=826802), which was resolved by backporting patches from upstream.

Steps to reproduce:

```bash
$ sudo xbps-install systemtap linux4.7-headers
[..snip..]
$ xbps-query -p pkgver systemtap
systemtap-3.0_1
$ xbps-query -p pkgver linux4.7
linux4.7-4.7.6_1
$ sudo stap -e 'probe oneshot { println("hello world") }'
In file included from /usr/share/systemtap/runtime/linux/runtime.h:209:0,
                 from /usr/share/systemtap/runtime/runtime.h:24,
                 from /tmp/stapJBFo3e/stap_f756c352e21eb4775cdc6b94c1907c5c_902_src.c:25:
/usr/share/systemtap/runtime/linux/access_process_vm.h: In function '__access_process_vm_':
/usr/share/systemtap/runtime/linux/access_process_vm.h:35:29: error: passing argument 1 of 'get_user_pages' makes integer from pointer without a cast [-Werror]
       ret = get_user_pages (tsk, mm, addr, 1, write, 1, &page, &vma);
                             ^
[..snip several more errors..]
cc1: all warnings being treated as errors
make[1]: *** [scripts/Makefile.build:289: /tmp/stapJBFo3e/stap_f756c352e21eb4775cdc6b94c1907c5c_902_src.o] Error 1
make: *** [Makefile:1458: _module_/tmp/stapJBFo3e] Error 2
WARNING: kbuild exited with status: 2
Pass 4: compilation failed.  [man error::pass4]
```

**Comment by ebfe** [comment]
If someone wants to work on this, this should be mostly be a matter of indentifying the right patches, putting them into `srcpkgs/systemtap/patches`, rebuilding systemtap `./xbps-src -f pkg systemtap`,  installing it `xi -f systemtap` and testing.

**Comment by chneukirchen** [comment]
Fixed in 98eb595.

**Comment by ackalker** [comment]
Thanks :)
